src/Cryptography/Sodium/Bindings/Signing.hs

```diff
@@ -65,30 +68,36 @@ import Foreign.C (CInt (..), CSize (..), CUChar (..), CULLong (..))
 -- Key pair generation --
 -------------------------
 
--- | This function randomly generates a secret key and a corresponding public key.
--- The public key is put into the @public key@ and the secret key into the @secret key@.
+-- | Randomly generate a secret key and a corresponding public key.
+--
+-- You can retrieve them by accessing the buffers passed as parameters
+-- to this function.
+--
+-- /See:/ [crypto_sign_keypair()](https://doc.libsodium.org/public-key_cryptography/public-key_signatures#key-pair-generation)
 --
 -- @since 0.0.1.0
 foreign import capi "sodium.h crypto_sign_keypair"
   cryptoSignKeyPair ::
-    -- | A pointer to an initialized cryptographic state. It has a length of 'cryptoSignPublicKeyBytes' bytes.
+    -- | A pointer to the buffer holding the public key. It has a length of 'cryptoSignPublicKeyBytes' bytes.
     Ptr CUChar ->
-    -- | A pointer to an initialized cryptographic state. It has a length of 'cryptoSignSecretKeyBytes' bytes.
+    -- | A pointer to the buffer holding the secret key. It has a length of 'cryptoSignSecretKeyBytes' bytes.
     Ptr CUChar ->
     -- | Return code is 0 on success, -1 on error.
     IO CInt
 
--- | This function derives a keypair (@secret key@ and @public key@) from a seed.
+-- | Derive a keypair (secret key and public key) from a seed.
 -- It is deterministic.
 --
+-- /See:/ [crypto_sign_seed_keypair()](https://doc.libsodium.org/public-key_cryptography/public-key_signatures#key-pair-generation)
+--
 -- @since 0.0.1.0
 foreign import capi "sodium.h crypto_sign_seed_keypair"
   cryptoSignSeedKeyPair ::
-    -- | A pointer to an initialized cryptographic state. It has a length of 'cryptoSignPublicKeyBytes'.
+    -- | A pointer to the buffer holding the public key. It has a length of 'cryptoSignPublicKeyBytes'.
     Ptr CUChar ->
-    -- | A pointer to an initialized cryptographic state. It has a length of 'cryptoSignSecretKeyBytes'.
+    -- | A pointer to the buffer holding the secret key. It has a length of 'cryptoSignSecretKeyBytes'.
     Ptr CUChar ->
-    -- | A pointer to an initialized cryptographic state. It has a length of 'cryptoSignSeedBytes'.
+    -- | A pointer to the seed. It has a length of 'cryptoSignSeedBytes'.
     Ptr CUChar ->
     -- | Return code is 0 on success, -1 on error.
```

Comment:
I would phrase it as "Returns 0 on success, -1 on error.", or even "0 on success, -1 on error.".